index.html

```diff
@@ -255,17 +290,20 @@ <h2>
         the following steps:
       </p>
       <ol>
-        <li>If the <a>incumbent settings object</a> is not a <a>secure
-        context</a>, then reject this <a>Navigator</a> object's <a>battery
-        promise</a> with a "<a>SecurityError</a>" <a>DOMException</a>, return
-        this <a>Navigator</a> object's <a>battery promise</a> and abort these
-        steps.
+        <li>If the <a>relevant settings object</a> of this <a>Navigator</a>
+        object is not a <a>secure context</a>, then reject this
+        <a>Navigator</a> object's <a>battery promise</a> with a
+        "<a>SecurityError</a>" <a>DOMException</a>, return this
+        <a>Navigator</a> object's <a>battery promise</a> and abort these steps.
         </li>
-        <li>If the <a>browsing context</a> is not a <a>top-level browsing
-        context</a>, then reject this <a>Navigator</a> object's <a>battery
-        promise</a> with a "<a>SecurityError</a>" <a>DOMException</a>, return
-        this <a>Navigator</a> object's <a>battery promise</a> and abort these
-        steps.
+        <li>If this <a>Navigator</a> object's <a>relevant global object</a>'s
```

Comment:
With feature policy integration, I think you can replace this entire paragraph with something like

> If this Navigator object's relevant global object's associated Document is not [allowed to use](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#allowed-to-use) the <code>battery</code> feature, then reject...

Because the default allowlist is `'self'`, that will automatically take care of the same-origin embed case, while allowing cross-origin usage only if explicitly enabled by the embedding document (which this paragraph still prohibits, I think)

Reply:
I pushed an update in an attempt to make use of "allowed to use". Please take a look. (I'm happy to see "allowed to use" abstracted out and reusable.)